projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -154,28 +174,40 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
      */
     public clearInput() {
         this.searchValue = null;
+        this.filterListData();
     }
 
     /**
      * @hidden @internal
      */
     public onCheckboxChange(eventArgs: IChangeCheckboxEventArgs) {
-        const selectedIndex = this.esf.listData.indexOf(eventArgs.checkbox.value);
+        const selectedIndex = this.displayedListData.indexOf(eventArgs.checkbox.value);
+        const selectAllBtn = this.displayedListData[0];
+
         if (selectedIndex === 0) {
-            this.esf.listData.forEach(element => {
+            this.displayedListData.forEach(element => {
+                if (element.value === this.addToCurrentFilter.value) { return; }
```

Comment:
This should work with comparing the objects themselves.
```suggestion
                if (element === this.addToCurrentFilter) { return; }
```